src/components/ProductRegistrationForm/Form.tsx

```diff
@@ -0,0 +1,272 @@
+}
+
+const INPUT_TEXT_VALIDATION_INDEX: number = 2
+const INPUT_NUMBER_VALIDATION_INDEX: number = 1
+const NUMBER_ONLY_PATTERN_VALUE = new RegExp(/^(?:[1-9]\d*|0)$/)
+const TEXT_ONLY_PATTERN_VALUE: RegExp = new RegExp(/^[\u0E00-\u0E7Fa-zA-Z' ,.'-]+$/i)
+const TEXT_AND_NUMBER_PATTERN_VALUE: RegExp = new RegExp(/^[\u0E00-\u0E7Fa-zA-Z0-9' ,.'-]+$/i)
+
+const Form = () => {
+  const [productImages, setProductImages] = useState<ImageListType>([])
+  const [ImageError, setImageError] = useState<ErrorsType>({})
+  const [selectedCategory, setSelectedCategory] = useState<string>('')
+  const [selectedDelivery, setSelectedDelivery] = useState<string>('')
+  const router = useRouter()
+  const { t } = useTranslation('product-registration')
+  const { register, handleSubmit, getValues, setValue, watch, errors } = useForm<FormInput>({
+    mode: 'onChange',
+    defaultValues: initialInputValue
+  })
+  // TODO: Fetch these options from database instead
+  const categoryOption: string[] = ['footwear', 'shirt', 'accessory'].map((option) => t(option))
+  const deliveryOption: string[] = ['postal', 'meetUp', 'both'].map((option) => t(option))
+  const hasProductImage: boolean = productImages.length > 0
+
+  useEffect(() => {
+    const handleWindowClose = (event: BeforeUnloadEvent) => {
+      if (hasProductImage) {
+        event.preventDefault()
+        event.returnValue = t('warningText')
+      }
+    }
+
+    const handleBrowseAway = () => {
+      if (hasProductImage && !window.confirm(t('warningText'))) {
+        router.events.emit('routeChangeError')
+        throw 'routeChange aborted.'
+      }
+    }
+
+    watch()
+    window.addEventListener('beforeunload', handleWindowClose)
+    router.events.on('routeChangeStart', handleBrowseAway)
+
+    return () => {
+      window.removeEventListener('beforeunload', handleWindowClose)
+      router.events.off('routeChangeStart', handleBrowseAway)
+    }
+  }, [productImages])
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    const { name, value } = event.target
+    setValue(name as keyof FormInput, value)
+  }
+
+  const handleFormSubmit = (value: FormInput) => {
+    // TODO: Send POST request to backend
+    setProductImages([])
+  }
+
+  const getErrorMessage = (inputKey: keyof FormInput, startValidationIndex: number): string => {
+    const errorMessage: string = errors[inputKey]?.message || ''
+    const isErrorDisplayed: boolean = getValues()[inputKey]?.length >= startValidationIndex
+
+    return errorMessage || isErrorDisplayed ? errorMessage : ''
+  }
+
+  const renderProductUpload = () => {
+    const renderErrorMessage = () => {
+      let errorMessage: string = ''
+
+      if (ImageError?.maxNumber && ImageError.maxFileSize) errorMessage = t('maxFilenNumberAndSizeExceed')
+      else if (ImageError?.maxNumber) errorMessage = t('maxFileNumberExceed')
+      else if (ImageError?.maxFileSize) errorMessage = t('maxFileSizeExceed')
+
```

Comment:
The best practice is that `if-else` should have a closure. How about this? It avoids using `let` though 🧐
```suggestion
const  message = () => {
    if () {
      return ...
    } else if {
      return ...
    } else {
      return ''
    }
  }
```

Reply:
Please see this [comment](https://github.com/Cobda/cobda-web/pull/55#discussion_r619605355)